compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5102,6 +5102,7 @@ private void checkActionInvocation(BLangInvocation.BLangActionInvocation aInv, B
         if (((varRef.symbol.tag & SymTag.ENDPOINT) != SymTag.ENDPOINT) && !aInv.async) {
             dlog.error(aInv.pos, DiagnosticCode.INVALID_ACTION_INVOCATION, varRef.type);
             this.resultType = symTable.semanticError;
+            aInv.symbol = symTable.notFoundSymbol;
```

Comment:
do we have to set the symbol explicitly here?

Reply:
This is to prevent the NPE here https://github.com/ballerina-platform/ballerina-lang/blob/5ac97156772ca44721774fe765c0b9d7bde2c3f2/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java#L2196

Reply:
can there be a situation where aInv already had the symbol and this change override that symbol ?

Reply:
For non error cases the symbol is set here, (bottom of the function) https://github.com/ballerina-platform/ballerina-lang/blob/a16b8a4e9a73c332607a19700e030ff8a9acfebd/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java#L5130

Reply:
ack